### ISSUE TITLE
BUG : SQLite open database fails on tvOS

### ISSUE THREAD

**Issue by xr1337** [rudderlabs/rudder-sdk-ios#440]
**Describe the bug**

SQLite support on tvOS fails due to file permissions. 

This is caused by tvOS  restrictions on document/library directory (   
https://github.com/rudderlabs/rudder-sdk-ios/blob/master-v2/Sources/Classes/Utility/RSUtils.swift#L27 )

The suggested fix by Apple is to create SQLite file in library/caches folder.

**To Reproduce**
Steps to reproduce the behaviour:

1. Initialise iOS SDK in tvOS app 
2. Run tvOS app on hardware (important as simulator does not have the same permission lockdown, thus does not throw the error)
3. Fire an event and Notice the error in console:

```
cannot open file at line 49259 of [1b37c146ee]
os_unix.c:49259: (2) open(/private/var/mobile/Containers/Data/Application/424D7D4E-556E-498B-9D45-36799A570570/Library/rl_persistence.sqlite) - No such file or directory
```

**Expected behavior**
No file system error should occur when running on tvOS

**Version of the _iOS_ SDK**
iOS SDK v2.4.2

** Initialization used **
```
let config: RSConfig = RSConfig(writeKey: getKey())
      .dataPlaneURL(getDataPlaneURL())
      .loglevel(.debug)
      .trackLifecycleEvents(true)
      .recordScreenViews(true)
    RSClient.sharedInstance().configure(with: config)
```

**Comment by desusai7** [comment]
@xr1337 we just released a fix for this issue in version 2.4.3 of the SDK, please try it out and let us know how it goes.

**Comment by xr1337** [comment]
Thank you @desusai7 

I tested version 2.4.3 and verified that the fix works 👍 .
Closing this issue.
